packages/cardpay-sdk/providers/json-rpc-provider.ts

```diff
@@ -14,8 +17,10 @@ function spelunk(value: any): null | { message: string; data: string } {
     return null;
   }
 
+  const isDataHexString = isHexString(value.data);
+
   // These *are* the droids we're looking for.
-  if (typeof value.message === 'string' && (isExpectedMessage(value.message) || isHexString(value.data))) {
+  if (isDataHexString || (isExpectedMessage(value.message) && isDataHexString)) {
```

Comment:
My wondering if we need this OR condition, or if just checking `isExpectedMessage(value.message) && isDataHexString` is enough

Reply:
Sokol doesn't throw error data with a hex value. The hex value is included in the error message.

`{
  message: 'Reverted 0x3756e7636b415b81342e52c75eb03c931941c450',
  data: undefined
}`

Reply:
I think we could have a condition like this. `isExpectedMessage(value.message) && (isDataHexString || value.data == undefined)`

Reply:
Done